### PR TITLE
fix: char counter update on emoji - [AP-169]

### PR DESCRIPTION
**Pull request by KemoPaw**
**What this PR does** 📖

- counter for char limit updates when adding emoji(s)
- emoji(s) are counted as only one "length" instead of two

**Which issue(s) this PR fixes** 🔨
https://satellite-im.atlassian.net/browse/AP-169

Fixes #

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Review by stavares843 — APPROVED** [review]
fixed ✅

File: components/tailored/core/chatbar/footer/Footer.vue [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
-      `${text.length}/${maxChars}`
-    }}</span>
+    <span :class="['charlimit', charlimit ? 'is-error' : 'is-normal']">
+      {{ `${calculateLength(ui.chatbarContent)}/${maxChars}` }}
+    </span>
   </div>
 </template>
 
 <script lang="ts">
 import Vue, { PropType } from 'vue'
```
Comment (WanderingHogan) [inline components/tailored/core/chatbar/footer/Footer.vue:11]: you can remove the imports for PropType, Friend

File: components/tailored/core/chatbar/footer/Footer.vue [hunk 2]
```diff
@@ -38,6 +35,17 @@ export default Vue.extend({
...
   },
+  methods: {
+    /**
+     * @method calculateLength
+     * @description Calculates the length of a message; and handles emojis to only be counted as one character length
+     * @returns new length of message
+     * @example calculateLength(ui.chatbarContent)
+     */
+    calculateLength(inputString: string) {
+      return _.toArray(inputString).length
```
Comment (WanderingHogan) [inline components/tailored/core/chatbar/footer/Footer.vue:46]: you imported the toArray method, you don't need the whole library, change `_.toArray(` to `toArray(`

**Review by WanderingHogan — CHANGES_REQUESTED** [review]
Couple minor updates to the Footer.vue file in the comments

File: components/tailored/core/chatbar/footer/Footer.vue (hunk 1 above)
Comment (KemoPaw) [reply components/tailored/core/chatbar/footer/Footer.vue:11]: @WanderingHogan Okay!

File: components/tailored/core/chatbar/footer/Footer.vue [hunk 3]
```diff
@@ -1,23 +1,20 @@
 <template>
   <div id="bottom-bar-container">
     <UiTypingIndicator v-if="typing" :users="usersTyping" />
-    <span :class="`charlimit ${charlimit ? 'is-error' : 'is-normal'}`">{{
-      `${text.length}/${maxChars}`
-    }}</span>
+    <span :class="['charlimit', charlimit ? 'is-error' : 'is-normal']">
+      {{ `${calculateLength(ui.chatbarContent)}/${maxChars}` }}
+    </span>
```
Comment (josephmcg) [inline components/tailored/core/chatbar/footer/Footer.vue:6]: I would pull this out into a computed to keep the markup cleaner and easier to read.

```
    <span :class="['charlimit', charlimit ? 'is-error' : 'is-normal']">
      {{ lengthCount }}
    </span>
```

```

computed: {
    ...mapState(['ui']),
    lengthCount() {
      return `${toArray(this.ui.chatbarContent).length}/${this.maxChars}`
    },
  },
```

**Review by josephmcg — CHANGES_REQUESTED** [review]

**Review by Jekrimo — APPROVED** [review]

**Review by josephmcg — APPROVED** [review]
🚀

**Review by stavares843 — APPROVED** [review]
neat! 🎉

**Comment by stavares843** [comment]
added missing dev review label again because hogan was requested for review

**Review by WanderingHogan — APPROVED** [review]

**Pull request merged**